server/services/transactionEventsController.go

```diff
@@ -34,6 +36,141 @@ func (controller *transactionEventsController) hasSignalErrorOfSendingValueToNon
 	return false
 }
 
+func (controller *transactionEventsController) extractEventsESDTOrESDTNFTTransfers(tx *transaction.ApiTransactionResult) ([]*eventESDT, error) {
+	rawEventsESDTTransfer := controller.findManyEventsByIdentifier(tx, transactionEventESDTTransfer)
+	rawEventsESDTNFTTransfer := controller.findManyEventsByIdentifier(tx, transactionEventESDTNFTTransfer)
+	rawEventsMultiESDTNFTTransfer := controller.findManyEventsByIdentifier(tx, transactionEventMultiESDTNFTTransfer)
+
+	rawEvents := make([]*transaction.Events, 0, len(rawEventsESDTTransfer)+len(rawEventsESDTNFTTransfer)+len(rawEventsMultiESDTNFTTransfer))
+	rawEvents = append(rawEvents, rawEventsESDTTransfer...)
+	rawEvents = append(rawEvents, rawEventsESDTNFTTransfer...)
+	rawEvents = append(rawEvents, rawEventsMultiESDTNFTTransfer...)
+
+	typedEvents := make([]*eventESDT, 0, len(rawEvents))
+
+	for _, event := range rawEvents {
+		numTopics := len(event.Topics)
+		if numTopics != 4 {
```

Comment:
we'll change `MultiESDTNFTTransfer` logs:
1. currently, if you transfer 2 tokens via `MultiESDTNFTTransfer` there will be 2 events with that identifier, each event containing 4 topics
2. starting with `rc/v1.6.0` (I think) there will be only one event generated and the topics will be `List<tokenID, nonce, value>+destination` (for example: `tkn0@00@01@tkn2@01@01@destination`)

You could implement the changes right now since it is backward compatible (the last topic is the destination, and you could split the list. currently, there will be a single entry in the list)

Reply:
Fixed :+1: